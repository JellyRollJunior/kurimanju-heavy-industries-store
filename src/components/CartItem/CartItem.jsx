import { productData } from '../../data/productData';
import styles from './CartItem.module.css';

const CartItem = ({ itemId, quantity, removeFromCart }) => {
  const { name, img, price } = productData.find((item) => item.id == itemId);

  return (
    <li className={styles.cartItem}>
      <img src={img} alt={name} />
      <div className={styles.data}>
        <h3 className={styles.name}>{name}</h3>
        <h4 className={styles.price}>{price}</h4>
        <h4 className={styles.quantity}>Quantity: {quantity}</h4>
        <button onClick={() => removeFromCart(itemId)} className={styles.removeButton}>Remove item</button>
      </div>
    </li>
  );
};

export { CartItem };
