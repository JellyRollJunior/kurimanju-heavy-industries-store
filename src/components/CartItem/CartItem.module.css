.cartItem {
  composes: secondaryColor from '../../styles/colors.module.css';
  display: flex;
  padding: 8px;
  height: 180px;
  border: 2px solid;
  border-radius: 4px;
  text-align: right;
}

.data {
    display: flex;
    flex-direction: column;
    align-items: right;
}

.name {
  composes: tertiaryColor from '../../styles/colors.module.css';
}

.price,
.quantity {
  composes: primaryColor from '../../styles/colors.module.css';
}

.removeButton {
  composes: submitBtn from '../Card/Card.module.css';
  max-width: 160px;
  margin: auto 0 0 10px;
}
