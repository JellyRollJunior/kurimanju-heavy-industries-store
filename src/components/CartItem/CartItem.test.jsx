import { render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import { CartItem } from './CartItem';
import userEvent from '@testing-library/user-event';

describe('Cart item component', () => {
  it('Renders product image, name, price, quantity, and remove button', () => {
    const { container } = render(<CartItem itemId={0} quantity={1} />);

    expect(container).toMatchSnapshot();
  });

  it('Calls removeFromCart(itemId) when remove button is click', async () => {
    const user = userEvent.setup();
    const itemId = 0;
    const removeFromCart = vi.fn();
    render(
      <CartItem itemId={itemId} quantity={1} removeFromCart={removeFromCart} />
    );

    await user.click(screen.getByRole('button', { name: 'Remove' }));

    expect(removeFromCart).toBeCalledWith(itemId);
  });
});
