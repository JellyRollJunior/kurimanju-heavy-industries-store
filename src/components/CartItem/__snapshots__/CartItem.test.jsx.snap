// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cart item component > Renders product image, name, price, quantity, and remove button 1`] = `
<div>
  <li
    class="_cartItem_61ad6d"
  >
    <img
      alt="X-4 Base"
      src="/src/assets/X4_base.png"
    />
    <div
      class="_data_61ad6d"
    >
      <h3
        class="_name_61ad6d"
      >
        X-4 Base
      </h3>
      <h4
        class="_price_61ad6d"
      >
        90 000 000
      </h4>
      <h4
        class="_quantity_61ad6d"
      >
        Quantity: 
        1
      </h4>
      <button
        class="_removeButton_61ad6d"
      >
        Remove
      </button>
    </div>
  </li>
</div>
`;
